packages/web/src/components/apps/use-files.tsx

```diff
@@ -183,6 +209,49 @@ export function FilesProvider({ app, channel, rootDirEntries, children }: Provid
     [app.id, openFolder],
   );
 
+  const deleteFolder = useCallback(
+    async (entry: DirEntryType) => {
+      await deleteDirectory(app.id, entry.path);
+      removeCachedFiles(filesRef.current, entry.path);
+      setOpenedFile((openedFile) => {
+        if (openedFile && openedFile.path.startsWith(entry.path)) {
+          return null;
+        }
+        return openedFile;
+      });
+      openedDirectoriesRef.current.delete(entry.path);
+      fileTreeRef.current = deleteNode(fileTreeRef.current, entry.path);
+      forceComponentRerender(); // required
+    },
+    [app.id],
+  );
+
+  const renameFolder = useCallback(
+    async (entry: DirEntryType, name: string) => {
+      const { data: newEntry } = await renameDirectory(app.id, entry.path, name);
+      // const oldFile = filesRef.current[entry.path];
```

Comment:
Do we need to keep this?

Reply:
This is what I need to fix before merging